src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php

```diff
@@ -550,4 +531,87 @@ protected function findOr404(RequestConfiguration $configuration)
 
         return $resource;
     }
+
+    /**
+     * @param RequestConfiguration $configuration
+     * @param ResourceInterface $resource
+     *
+     * @return array
+     */
+    protected function getShowTemplateData(RequestConfiguration $configuration, ResourceInterface $resource)
+    {
+        return [
+            'configuration' => $configuration,
+            'metadata' => $this->metadata,
+            'resource' => $resource,
+            $this->metadata->getName() => $resource,
+        ];
+    }
+
+    /**
+     * @param RequestConfiguration $configuration
+     * @param mixed $resources
+     *
+     * @return array
+     */
+    protected function getIndexTemplateData(RequestConfiguration $configuration, $resources)
+    {
+        return [
+            'configuration' => $configuration,
+            'metadata' => $this->metadata,
+            'resources' => $resources,
+            $this->metadata->getPluralName() => $resources,
+        ];
+    }
+
+    /**
+     * @param RequestConfiguration $configuration
+     * @param ResourceInterface $newResource
+     * @param FormInterface $form
+     *
+     * @return array
+     */
+    protected function getCreateTemplateData(
```

Comment:
Ok, so there are two approaches:
1. ``provideFormTemplateData`` for create and update action
2. ``provideCreate\UpdateFormTemplateData`` which uses ``provideFormTemplateData`` for templates data separation

👍  for 1, 🎉  for 2, ready, set, go! 😄